_overviews/scala3-macros/tutorial/macros.md

```diff
@@ -131,28 +131,30 @@ Other types can also work if a `ToExpr` is implemented for it, we will [see this
 
 ### Extracting Values from Expressions
 
-The second method we use in the implementation of `powerCode` is `Expr[T].valueOrError`, which has an effect opposite to `Expr.apply`.
+The second method we use in the implementation of `powerCode` is `Expr[T].valueOrAbort`, which has an effect opposite to `Expr.apply`.
 It attempts to extract a value of type `T` from an expression of type `Expr[T]`.
 This can only succeed, if the expression directly contains the code of a value, otherwise, it will throw an exception that stops the macro expansion and reports that the expression did not correspond to a value.
 
-Instead of `valueOrError`, we could also use the `value` operation, which will return an `Option`.
+Instead of `valueOrAbort`, we could also use the `value` operation, which will return an `Option`.
 This way we can report the error with a custom error message.
 
 ```scala
   ...
+  import quotes.reflect.*
```

Comment:
Where does `quotes` come from?

Reply:
This is a little bit of magic that I don't understand myself, but it's coming from `using Quotes`
How does that work here exactly I don't know. If the example would like this

```scala
  x: Expr[Double],
  n: Expr[Int]
)(using quotes: Quotes): Expr[Double] =
    import quotes.reflect.*
```
it would be more explicit, 
but it also works as 

```scala
  x: Expr[Double],
  n: Expr[Int]
)(using Quotes): Expr[Double] =
    import quotes.reflect.*
```
But definitely `report` is coming from `quotes: Quotes`

Reply:
Is the fully qualified import `scala.quoted.quotes.*` then? As I see it seems to be this definition: https://scala-lang.org/api/3.x/scala/quoted.html#quotes-1e0

Reply:
I added in https://github.com/scala/docs.scala-lang/pull/2539/commits/0423507de3df93dedfd00fc1320d13102454c2fe some explanation